mod.Lua

```diff
@@ -1,22 +1,20 @@
+local total_xp = 0 -- Create variables outside the hook, so we dont recreate them every hookcall
+local notification_text = ""
 
-local previous_total_xp = 0
-Hooks:PreHook(ExperienceManager, "mission_xp_award", "xp_notifier_before_xp_increase", function ()
-	previous_total_xp = get_current_total_xp()
-end)
+function xp_string(n)
+	if managers.experience ~= nil then -- Check for null/nil so we dont crash
+	   return managers.experience:experience_string(n)
+	else 
+	   return tostring(n)
+	end
+end
 
-Hooks:PostHook(ExperienceManager, "mission_xp_award", "xp_notifier_after_xp_increase", function ()
-	local new_total_xp = get_current_total_xp()
-	local xp_added = new_total_xp - previous_total_xp
+Hooks:PostHook(ExperienceManager, "mission_xp_award", "xp_notifier_after_xp_increase", function(self, amount)
+	total_xp = total_xp + amount
 	
-	local notification_text = xp_string(xp_added) .. " XP gained (Total XP: " .. xp_string(new_total_xp) .. ")"
-	managers.hud:show_hint({ text = notification_text })
+	notification_text = xp_string(amount) .. " XP gained (Total XP: " .. xp_string(total_xp) .. ")"
+	
+	if managers.hud ~= nil then
+	   managers.hud:show_hint({ text = notification_text })
+	end
 end)
-
-function get_current_total_xp()
-	local num_alive_players = managers.network:session():amount_of_alive_players()
-	return managers.experience:get_xp_dissected(true, num_alive_players, true)
```

Comment:
`managers.experience:get_xp_dissected` takes XP bonuses into account, such as:
- number of alive players (second parameter)
- whether the player is not in custody (third parameter, just realized this isn't always `true`)
- current difficulty (calculated inside the function itself)